content/docs/user-guide/experiment-management/experiments-overview.md

```diff
@@ -1,10 +1,29 @@
 # DVC Experiments Overview
 
-DVC Experiments are captures automatically by DVC when you [run them].
+DVC Experiments are captures automatically by DVC when you [run them]. Each
+experiment creates and tracks a project variation based on the changes in your
+<abbr>workspace</abbr>. Experiments preserve the latest commit in the current
+branch (Git `HEAD`) as their parent or _baseline_.
+
+<details>
+
+### ⚙️ How does DVC track experiments?
+
+Experiments are custom [Git references] (found in `.git/refs/exps`) with a
+single commit based on `HEAD` (not checked out by DVC). Note that these commits
+are not pushed to Git remotes by default (see `dvc exp push`).
+
+</details>
+
+Experiments will have an auto-generated ID like `exp-bfe64` by default. A custom
+name can be given instead (using the `--name`/`-n` option of `dvc exp run`).
+
+> ID or name can be used to reference experiments with `dvc exp` subcommands.
```

Comment:
@iesahin I'm not 100% sure this is right or consistent with the language you've used in other guides. Are `exp-abcde` called "IDs" or "auto-generated names"? Are the Git hashes called IDs instead? The hashes don't exist for queued experiments so it may be confusing to call those IDs.

We should make sure this terminology makes sense and is consistent 🙂 - let's figure it out here and I'll apply everywhere else.

Reply:
I was using ID and name interchangeably to tell `exp-abc123` like names, which can be specified with `--name` option to `dvc  exp run`. Unfortunately, there are places that we need to tell `refs/exps/abc123eeefff333/` IDs as commit IDs. 

You're right that we should avoid ID to denote the former, and call the latter _commit ID_ instead. Could this be the naming convention?

Reply:
"commit ID" may also be confusing though. We already have a term for that I think? I.e. SHA or commit hash.

Maybe we only need "SHA/hash" and "name"? Avoid "ID" completely. WDYT?

Reply:
Could we put "experiment name" to the glossary? When we mention it for the first time, we can link to it. 

I'll try to avoid ID completely. 

Commit hash = SHA-256 of Git commits
Experiment name = `exp-123ab` like names in experiments.

Reply:
Can we have a wiki page about these notes, and review time to time? @jorgeorpinel